### PR TITLE
hevm exec --trace

### PR DESCRIPTION
**Pull request by rainbreak**
Adds a --trace flag to `hevm exec`, which will dump the execution trace.

Currently just the basic functionality, future prs will add more tracing features.

### CODE REVIEW

File: src/hevm/hevm-cli/hevm-cli.hs [hunk 1]
```diff
@@ -513,6 +524,10 @@ launchExec cmd = do
   case optsMode cmd of
     Run -> do
       vm' <- execStateT (interpret fetcher Nothing . void $ EVM.Stepper.execFully) vm1
+      if (trace cmd) then
+        hPutStr stderr (showTraceTree dapp vm')
+      else
+        pure ()
```
Comment (MrChico) [inline src/hevm/hevm-cli/hevm-cli.hs:530]:
```suggestion
      when (trace cmd) $ hPutStr stderr (showTraceTree dapp vm')
```

File: src/seth/libexec/seth/seth-run-tx [hunk 2]
```diff
@@ -40,6 +40,7 @@ opts+=(--chainid   "${ETH_CHAINID:-$(seth rpc eth_chainId | seth --to-dec)}")
 
 [[ "$HEVM_STATE" ]] && opts+=(--state "$HEVM_STATE")
 [[ "$HEVM_DEBUG" ]] && opts+=(--debug)
+[[ "$HEVM_DEBUG" ]] && opts+=(--trace)
```
Comment (MrChico) [inline src/seth/libexec/seth/seth-run-tx:43]:
```suggestion
[[ "$HEVM_TRACE" ]] && opts+=(--trace)
```

File: src/hevm/hevm-cli/hevm-cli.hs (hunk 1 above)
Comment (rainbreak) [reply src/hevm/hevm-cli/hevm-cli.hs:530]: TIL `when` :)

**Comment by MrChico** [comment]
This is a good start. `--trace` could also be added to hevm symbolic, just to keep things in par

**Comment by rainbreak** [comment]
Ah yeah, will do that here.

**Comment by rainbreak** [comment]
Actually, will skip `symbolic --trace` here as it's not entirely simple. Are you happy to merge as-is @MrChico ?

**Pull request merged**